build_tools/cmake/build_host_tools.sh

```diff
@@ -34,13 +34,24 @@ fi
 mkdir -p "${INSTALL_DIR}"
 
 # Configure, build, install.
-"${CMAKE_BIN}" -G Ninja -B "${BUILD_DIR}" \
-  -DCMAKE_INSTALL_PREFIX="$(realpath ${INSTALL_DIR})" \
-  -DIREE_ENABLE_LLD=ON \
-  -DIREE_ENABLE_ASSERTIONS="${IREE_ENABLE_ASSERTIONS}" \
-  -DIREE_BUILD_COMPILER=ON \
-  -DIREE_BUILD_TESTS=OFF \
-  -DIREE_BUILD_SAMPLES=OFF \
-  "${ROOT_DIR}"
+declare -a CMAKE_ARGS=(
+  "-G" "Ninja"
+  "-B" "${BUILD_DIR}"
+
+  "-DCMAKE_INSTALL_PREFIX=$(realpath ${INSTALL_DIR})"
+  "-DIREE_ENABLE_LLD=ON"
+  "-DIREE_ENABLE_ASSERTIONS=${IREE_ENABLE_ASSERTIONS}"
+  "-DIREE_BUILD_COMPILER=ON"
+  "-DIREE_BUILD_TESTS=OFF"
+  "-DIREE_BUILD_SAMPLES=OFF"
+
+  # Enable CUDA compiler and runtime builds unconditionally. Our CI images all
+  # have enough deps to at least build CUDA support and compile CUDA binaries
+  # (but not necessarily test on real hardware).
+  "-DIREE_HAL_DRIVER_CUDA=ON"
+  "-DIREE_TARGET_BACKEND_CUDA=ON"
```

Comment:
Not sure if we should enable the CUDA backend in this script or there is a way to get host binaries from build_all.sh

Reply:
I think it's reasonable to enable CUDA here. These tools exist for usage of other jobs and having cuda tools is helpful. Splitting this out into a bunch of different jobs is possible, but seems worse than having it all in one (for now)

Reply:
The Linux pipeline uses this script to build host tools, then the tools are used to build CUDA benchmark suites.
https://github.com/iree-org/iree/blob/37dc407dce679385173bedcc30b66bc12a97b613/build_tools/buildkite/cmake/linux/pipeline.yml#L25

Reply:
Ooooh it's already using it I see. I was ctrl+f searching for it elsewhere in your PR

Reply:
+1 seems reasonable. A few other builds run this script, so we might want to keep an eye on if the CUDA backend / HAL driver substantially increase build time.